### ISSUE TITLE
Improve basionym chain normalizer performance

### ISSUE THREAD

**Issue by mdoering** [CatalogueOfLife/backend#941]
WoRMS Mollusca ColDP with ~200.000 names takes 13h to resolve basionym chains.
The query seems to be very bad, this should not take so much time in NeoDB:

>   | 2020-12-24 04:03:46.505 | INFO | - | - | life.catalogue.importer.Normalizer | 56068 basionym chains resolved
>   | 2020-12-23 15:21:17.136 | INFO | - | - | life.catalogue.importer.Normalizer | Synonym relations cleaned up. 0 hasParent relations deleted,9 isParentOf relations deleted, 195 isParentOf rels moved from synonym to accepted
>   | 2020-12-23 15:21:17.136 | INFO | - | - | life.catalogue.importer.Normalizer

**Comment by gdower** [comment]
For Name.basionymID, the CoLDP documentation says, "[Contrary to the strict basionym definition it is recommended to populate this field also for original names which should point to itself.](https://github.com/catalogueoflife/coldp#basionymid)" WoRMS followed that and those names get flagged as chained basionyms:

https://data.catalogueoflife.org/dataset/2251/verbatim?issue=chained%20basionym

For example, the first record has and is flagged with the chained basionym issue:

column | row
--|--
ID | urn:lsid:marinespecies.org:taxname:1252865
basionymID | urn:lsid:marinespecies.org:taxname:1252865

**Comment by mdoering** [comment]
A bug I will fix, please keep that mapping in WoRMS!

**Comment by gdower** [comment]
Maybe it is just a bad link in ChecklistBank? The HTML code is:

```
<a href="urn:lsid:marinespecies.org:taxname:1252865" target="_blank">urn:lsid:marinespecies.org:taxname:1252865</a>
```

So it's missing the full URL.

**Comment by gdower** [comment]
If it's not all that difficult to switch off the basionym chain normalizer if the ID == basionymID then it would be good to prioritize this because it might be slowing down WoRMS development and [blocks the 1 importer queue thread on dev](https://github.com/CatalogueOfLife/checklistbank/issues/816#issuecomment-788226477). If there isn't a quick fix, then we can probably get by.

**Comment by mdoering** [comment]
basionymID=taxonID is not causing any problems, this gets entirely ignored as reproduced in a new test.
There must be sth else that causes this in the data. The `chained basionym` flag gets applied to both sides of the relation, so it can be any other name record pointing there too. 

If you [search the verbatim name and name_rel data for the above ID](https://data.catalogueoflife.org/dataset/2251/verbatim?col%3AbasionymID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3AnameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3AnameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3AnameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3AnameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3AnameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&col%3ArelatedNameID=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&q=urn%3Alsid%3Amarinespecies.org%3Ataxname%3A1252865&termOp=OR&type=col%3AName&type=col%3ANameRel), various relations show up:

<img width="1599" alt="Screenshot 2021-03-01 at 21 57 01" src="https://user-images.githubusercontent.com/327505/109557987-201a0480-7ad9-11eb-821c-fb9e29a2fb1a.png">

 - NameRel 1252864 -basionym-> 1252865
 - Name 1252865 -basionymID=1252865
 - NameRel 1252865 -basionym-> 1252865
 - Name 1252864 -basionymID=1252865 
 - 
 Both Name.basionymID and NameRel.relatedNameID with type=basionym is a directed relation "hasBasionym" and points to the basionym. The data appears to be correct in that sense. The superflous name relation record should probably not be present, but it should also not be harmful and simply be ignored. Maybe the duplicate NameRel and Name.basionymID relation is causing a problem here, but it should not.

**Comment by mdoering** [comment]
Yes, thats it. I'll fix this and hopefully this improves the performance too.

**Comment by mdoering** [comment]
The chained basionym issue should be gone now and hopefully performance much better.
I am curious how Molluscs performs with a new deploy

**Comment by mdoering** [comment]
deployed to dev

**Comment by mdoering** [comment]
Please test and reopen if its not good enough, I will forget to close this issue otherwise...

**Comment by gdower** [comment]
Mollusca still takes 12 hours to import, although this is not a high priority to fix.
